modules/core/src/main/java/org/apache/synapse/message/store/impl/rabbitmq/RabbitMQProducer.java

```diff
@@ -36,147 +36,186 @@
+            Thread.currentThread().interrupt();
+        } catch (TimeoutException | IOException e) {
+            String errorMsg = getId() + ". Ignored MessageId: " + synCtx.getMessageID() + ". " +
+                    "Could not store message to store [" + store.getName() + "]. " +
+                    "Error:" + e.getLocalizedMessage();
+            log.error(errorMsg, e);
+        }
+        if (log.isDebugEnabled()) {
+            log.debug(getId() + ". Stored MessageId: " + synCtx.getMessageID());
+        }
+        store.enqueued();
+        return result;
+    }
+
+    /**
+     * Serialize the message to store in the queue
+     *
+     * @param storableMessage the {@link StorableMessage} object
+     * @return serialize message as byte array
+     * @throws IOException
+     */
+    private byte[] serializeMessage(StorableMessage storableMessage) throws IOException {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        ObjectOutput objectOutput = new ObjectOutputStream(outputStream);
+        objectOutput.writeObject(storableMessage);
+        return outputStream.toByteArray();
+    }
+
+    /**
+     * Build AMQP basic properties from the message context
+     *
+     * @param synCtx          the {@link MessageContext} object
+     * @param storableMessage the {@link StorableMessage} object
+     * @return AMQP basic properties
+     */
+    private AMQP.BasicProperties getBasicProperties(MessageContext synCtx, StorableMessage storableMessage) {
+        AMQP.BasicProperties.Builder builder = new AMQP.BasicProperties().builder();
+        builder.messageId(synCtx.getMessageID());
+        builder.deliveryMode(MessageProperties.MINIMAL_PERSISTENT_BASIC.getDeliveryMode());
+        builder.priority(storableMessage.getPriority(DEFAULT_PRIORITY));
+        return builder.build();
+    }
+
+    /**
+     * Perform basic publish
+     *
+     * @param channel         the channel
+     * @param exchangeName    the exchange to publish the message to
+     * @param routingKey      the routing key
+     * @param basicProperties other properties for the message
+     * @param messageBody     the message body
+     * @throws IOException
+     */
+    private void publishMessage(Channel channel, String exchangeName, String routingKey,
+                                AMQP.BasicProperties basicProperties, byte[] messageBody) throws IOException {
+        if (StringUtils.isNotEmpty(exchangeName)) {
+            channel.basicPublish(exchangeName, routingKey, basicProperties, messageBody);
+        } else {
+            channel.basicPublish("", routingKey, basicProperties, messageBody);
+        }
+    }
+
+    /**
+     * Used to close the channel opened in this object instance.
+     * This should be called after the end of each call on storeMessage method
+     * But instead of this, try with resources will close the channel
+     */
+    @Override
+    public boolean cleanup() {
+        return true;
```

Comment:
Shouldn't we close the created rabbitmq connection here?

Reply:
No. This method should be called after the end of each call on storeMessage method. But it doesn't require since try with resources used.